src/redux/actions/loadUser.ts

```diff
@@ -2,11 +2,32 @@ import {Dispatch} from "react";
 import AsyncStorage from "@react-native-async-storage/async-storage";
 import axios, {AxiosError} from "axios";
 import {URI} from "../../URI";
+import {userLoadFailed, userLoadRequest, userLoadSuccess} from "../types/types";
 
-export const loadUser = () => async (dispatch: Dispatch<any>) => {
+interface LoadUserRequestAction {
+    type: 'userLoadRequest';
+}
+
+interface LoadUserSuccessAction {
+    type: 'userLoadSuccess';
+    payload: { user: string, token: string | null };
+}
+
+interface LoadUserFailedAction {
+    type: 'userLoadFailed';
+    payload: string;
+}
+
+type LoadUserAction =
+    | LoadUserRequestAction
+    | LoadUserSuccessAction
+    | LoadUserFailedAction;
+
+
+export const loadUser = () => async (dispatch: Dispatch<LoadUserAction>) => {
     try {
         dispatch({
-            type: 'userLoadRequest',
+            type: userLoadRequest,
         });
 
         const token = await AsyncStorage.getItem('token');
```

Comment:
Also, AsyncStorage isn't a safe place for our secrets. We should not keep our tokens here. See https://reactnative.dev/docs/security for alteratives